src/actions.js

```diff
@@ -75,76 +74,71 @@ export function addCheckResult(
 // ASYNC (THUNK) ACTIONS.
 
 // Fetching the statuses.
-export function requestStatus(version: ?string): ThunkAction<void> {
+export function requestStatus(version: ?string) {
   const notifyChanges = (checkTitle, status) => {
     if (Notification.permission === 'granted') {
       new Notification(`${checkTitle} : status changed (${status}).`);
     }
   };
 
-  return function(dispatch: Dispatch, getState: GetState) {
+  return async function(dispatch: Dispatch, getState: GetState) {
     const versionToCheck = version || getState().version;
     if (!versionToCheck) {
       return;
     }
     // Save previous results so we can check if something changed.
     const prevResults = getState().checkResults;
     dispatch(setVersion(versionToCheck));
-    getReleaseInfo(versionToCheck)
-      .then(releaseInfo => {
-        dispatch(updateReleaseInfo(releaseInfo));
-        releaseInfo.checks.map((check: CheckInfo) => {
-          return checkStatus(check.url).then(result => {
-            // Detect if the result changed, and notify!
-            const prevResult = prevResults[check.title];
-            if (prevResult && prevResult.status !== result.status) {
-              notifyChanges(check.title, result.status);
-            }
-            dispatch(addCheckResult(check.title, result));
-          });
-        });
-      })
-      .catch((err: string) => {
-        console.error(
-          `Failed getting the release info for ${versionToCheck}`,
-          err,
-        );
+    try {
+      const releaseInfo = await getReleaseInfo(versionToCheck);
+      dispatch(updateReleaseInfo(releaseInfo));
+      releaseInfo.checks.map(async (check: CheckInfo) => {
```

Comment:
It would be nice to use map to build the list of promise and then do a `Promise.all` on the list.
Maybe we could use this: https://hunterloftis.github.io/awaiting/#map

Reply:
According to https://stackoverflow.com/a/31414472/186202 what you are doing here is already firing all the promise.
We don't need Promise.all here because we don't have other actions that we want to do after all of the Promise returned.

Reply:
Not sure this answer talks about the exact case i have here, which is `map`ing on `await`s

Reply:
It looks to me like your `map` call is going to produce an array of Promises. I'd expect these promises to all happen in parallel and immediately, even if you don't await them. And like @Natim says, since we don't need the result of those promises, we probably don't need to await them. However, the link @Natim posted suggests that it might be good to await those promises so that errors are surfaced. Otherwise, we have a rejection which isn't being handled, which means it isn't reported anywhere. [This MDN page](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onunhandledrejection) says that Firefox would like to one day have an event that reports unhandled rejections, but it isn't implemented yet.

Reply:
I'll modify the code to use Promise.all then ;) Thanks for the feedback!